backend_kvm.go

```diff
@@ -0,0 +1,136 @@
+// +build linux
+// +build amd64
+
+package fakemachine
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+type KvmBackend struct {}
+
+func init() {
+	RegisterBackend(KvmBackend{})
+}
+
+func (b KvmBackend) Name() string {
+	return "kvm"
+}
+
+func (b KvmBackend) Supported() (bool, error) {
+	if _, err := os.Stat("/dev/kvm"); err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+func (b KvmBackend) InitScriptModules() []string {
+	return []string{"virtio_pci", "virtio_console", "9pnet_virtio", "9p"}
+}
+
+func (b KvmBackend) GetExtraVolumes() []mountPoint {
+	return nil
+}
+
+func (b KvmBackend) MountOptions(mount mountPoint) (fstype string, options []string, device string, mountpoint string) {
+	fstype = "9p"
+	options = []string{"trans=virtio", "version=9p2000.L", "cache=loose", "msize=262144"}
+	device = mount.label
+	mountpoint = mount.machineDirectory
+	return
+}
+
+func (b KvmBackend) NetworkdMatch() string {
+	return "e*"
+}
+
+func (b KvmBackend) GetImagePath(img image) string {
+	return fmt.Sprintf("/dev/disk/by-id/virtio-%s", img.label)
```

Comment:
depending on the backend just the image may not be enough; virtio is nice in that it follows the labels (if formatted properly) but others may not have that and would need to depend on e.g. the index of the virtual disk.. Which comes back to the comment further down about creating a stateful instance first.

Reply:
further on the image struct was modified to store the index of the virtual disk ;-)

Reply:
Yeah so that leaks more implementation details in common structures *and* adds the assumption that the backend will start in a way such that the virtual machine follows those indices